pkg/pipeline/sink/uploader/s3.go

```diff
@@ -55,7 +62,42 @@ func newS3Uploader(conf *livekit.S3Upload) Uploader {
 	return u
 }
 
+func (u *S3Uploader) getBucketLocation(storageFilepath string) (string, error) {
+	u.awsConfig.Region = aws.String(getBucketLocationRegion)
+
+	sess, err := session.NewSession(u.awsConfig)
+	if err != nil {
+		return "", err
+	}
+
+	req := &s3.GetBucketLocationInput{
+		Bucket: u.bucket,
+	}
+
+	svc := s3.New(sess)
+	resp, err := svc.GetBucketLocation(req)
+	if err != nil {
+		return "", errors.ErrUploadFailed(storageFilepath, err)
+	}
+
+	if resp.LocationConstraint == nil {
+		return "", errors.ErrUploadFailed(storageFilepath, errors.New("bucket location was nil"))
+	}
+
+	return *resp.LocationConstraint, nil
+}
+
 func (u *S3Uploader) Upload(localFilepath, storageFilepath string, outputType types.OutputType) (string, int64, error) {
+	if u.awsConfig.Region == nil {
```

Comment:
this should happen in newS3Uploader

Reply:
I was trying to avoid doing blocking I/O in the "new" routine, particularly since AWS will charge the owner of the bucket for the request. Any reason to make it early?

Reply:
to fail before the egress runs instead of after